src/core.js

```diff
@@ -425,6 +425,16 @@ jQuery.extend({
 	support: support
 });
 
+// jsHint would error on this code due to the Symbol not being defined in ES5.
+// Defining this global in .jshintrc would create a danger of using the global
+// unguarded in another place, it seems safer to just disable jsHint for these
+// three lines.
+/* jshint ignore: start */
+if ( typeof Symbol === "function" && Symbol.iterator ) {
```

Comment:
> The "worst outcome" here could include a discoverable string property on the jQuery prototype, that people may rely on in browsers without native `Symbol`, and thus their code will break in newer browsers.

@ljharb What would such a reliance look like? `for ( el of $collection )` won't work without a valid `Symbol.iterator`, but even if someone loads jQuery on top of an incomplete polyfill, there's nothing for them to do with a property named `"undefined"` or `"Symbol (iterator)"` or `"@@iterator"` if it has an `undefined` or other falsy value.